File: src/RussianSitesStatus/Models/ViewModels/Site.cs
using RussianSitesStatus.Models.ViewModels;

namespace RussianSitesStatus.Models;

public class Site: BaseModel
{    
    public string Name { get; set; }
    public string Status { get; set; }
    public double? Uptime { get; set; }
}

public class SiteDetails : Site
{
    public List<Server> Servers { get; set; }    
    public DateTime LastTestedAt { get; set; }
}

public class Server
{
    public string Region { get; set; }
    public string RegionCode { get; set; }
    public string Status { get; set; }
    public int StatusCode { get; internal set; }
    public int SpentTimeInSec { get; internal set; }
}

File: UI/src/app/models/site.ts
export class Site {
    id: number;
    name: string;
    status: string;
}

export class SiteDetails extends Site {
    servers: Server[];
    lastTestedAt: string;
}

export class Server {
    region: string;
    regionCode: string;
    spentTimeInSec: number;
    status: string;
    statusCode: number;
}
